operator/internal/common/extensions/secret_reconciliator.go

```diff
@@ -134,8 +135,8 @@ func (r *SecretReconciliator) updateExisting(ctx context.Context, secret *coreV1
 		needsUpdate = true
 	}
 
-	labels, needsLabelUpdate := commonLabels.WithDefaults(secret.Labels)
-	secret.Labels = labels
+	desiredLabels, needsLabelUpdate := commonLabels.WithDefaults(secret.Labels)
+	secret.Labels = desiredLabels
```

Comment:
Maybe rename `desiredLabels` to `labels` here otherwise it conflicts with the function parameter name

Reply:
Also I think 
```
desiredLabels, needsLabelUpdate := commonLabels.WithDefaults(secret.Labels)
```
will not achieve what you want. You probably need to add a function in `commonLabels` to merge labels, unless there is already an existing function for that.